src/include/btree.i

```diff
@@ -1220,11 +1220,19 @@ __wt_leaf_page_can_split(WT_SESSION_IMPL *session, WT_PAGE *page)
     WT_BTREE *btree;
     WT_INSERT *ins;
     WT_INSERT_HEAD *ins_head;
+    WT_REF *ref;
     size_t size;
     int count;
 
     btree = S2BT(session);
 
+    ref = page->pg_intl_parent_ref;
```

Comment:
This seems weird. Is this reference to the "parent" and not this page.
Maybe pass the ref to this function and derive the page from the ref.

Reply:
Good catch. I just read through the documentation and this would have been only valid in case of internal page.